### ISSUE TITLE
Misguiding error during `helmfile apply/template`

### ISSUE THREAD

**Issue by andrewnazarov** [roboll/helmfile#514]
I wanted to get to know if several prepare hooks inside a certain release worked. So I defined them like that: 

```
- name: cert-manager
    namespace: cert-manager
    chart: stable/cert-manager
    version: ~0.6.6
    hooks:
      - events: ["prepare"]
        command: "kubectl"
        args: ["apply", "-f", "https://raw.githubusercontent.com/jetstack/cert-manager/v0.6.0/deploy/manifests/00-crds.yaml"]
      - events: ["prepare"]
        command: "kubectl"
        args: ["label", "namespace", "cert-manager", "certmanager.k8s.io/disable-validation=\"true\""]
      - events: ["cleanup"]
        command: "kubectl"
        args: ["apply", "-f", "cert-manager/cluster-issuer.yaml"]
```
and ran `helmfile template` afterwards. It seems this triggers the error.

But instead of `cert-manager` I've got an error message related to a different release:
```
helmfile.yaml: basePath=/Users/andrewnazarov/Repo/iris-4/infra/init
err: release "nosqlclient" in "helmfile.yaml" failed: hook[kubectl]: command `kubectl` failed: exit status 1
hook[kubectl]: command `kubectl` failed: exit status 1
```
Worth to mention, that `nosqlclient` is the last release in the definition file. There are no hooks inside `nosqlclient` release:

```
- name: nosqlclient
    namespace: service
    chart: ./nosqlclient
    version: ~0.1.0
    values:
    - "./chartmuseum/values.yaml"
```

Btw, what is the propper way to create several hooks of the same type?

**Comment by andrewnazarov** [comment]
`helmfile version v0.47.0`

**Comment by mumoshu** [comment]
Thanks for reporting.

It does seem like something is going wrong! I have no idea any problem other than data race causes it.

**Comment by mumoshu** [comment]
@andrewnazarov This should be fixed via #614. Thanks again for reporting!
